pkg/apis/v1alpha1/nsxserviceaccount_types.go

```diff
@@ -0,0 +1,85 @@
+/* Copyright © 2022 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: Apache-2.0 */
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// NSXServiceAccountSpec defines the desired state of NSXServiceAccount
+type NSXServiceAccountSpec struct {
+	VPCName string `json:"vpcName,omitempty"`
+}
+
+type NSXProxyEndpointAddress struct {
+	Hostname string `json:"hostname,omitempty"`
+	IP       string `json:"ip,omitempty"`
+}
+
+type NSXProxyProtocol string
+
+const (
+	NSXProxyProtocolTCP NSXProxyProtocol = "TCP"
+)
+
+type NSXProxyEndpointPort struct {
+	Name     string           `json:"name,omitempty"`
+	Port     uint16           `json:"port,omitempty"`
```

Comment:
what is Port is empty, does it mean all ports?

Reply:
Empty port is "0", and this is not a valid value for a NSXProxy.